### PR TITLE
[GOBBLIN-506] Add job tags support for gobblin cluster

### PR DESCRIPTION
**Pull request by yukuai518**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-506

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
   Add tag binding from GobblinTaskRunner when helixManager first connects
   Add tag binding from jobConfig.Builder when cluster manager submits job to Helix
   Some refactoring from ClusterIntegrationTest and TaskFactoryBuilder inside of GobblinTaskRunner.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
   Added testJobWithTag in ClusterIntegrationTest.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by yukuai518** [comment]
@htran1 Can you help review?

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterConfigurationKeys.java [hunk 1]
```diff
@@ -63,6 +63,8 @@
   // Should job be executed in the scheduler thread?
   public static final String JOB_EXECUTE_IN_SCHEDULING_THREAD = GOBBLIN_CLUSTER_PREFIX + "job.executeInSchedulingThread";
   public static final boolean JOB_EXECUTE_IN_SCHEDULING_THREAD_DEFAULT = true;
+  public static final String HELIX_JOB_TAGS_KEY = GOBBLIN_CLUSTER_PREFIX + "helix.job.tags";
+  public static final String HELIX_INSTANCE_TAGS_KEY = GOBBLIN_CLUSTER_PREFIX + "helix.instance.tags";
```
Comment (htran1) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterConfigurationKeys.java]: helix.instance.tags -> helixInstanceTags

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterConfigurationKeys.java [hunk 2]
```diff
@@ -63,6 +63,8 @@
   // Should job be executed in the scheduler thread?
   public static final String JOB_EXECUTE_IN_SCHEDULING_THREAD = GOBBLIN_CLUSTER_PREFIX + "job.executeInSchedulingThread";
   public static final boolean JOB_EXECUTE_IN_SCHEDULING_THREAD_DEFAULT = true;
+  public static final String HELIX_JOB_TAGS_KEY = GOBBLIN_CLUSTER_PREFIX + "helix.job.tags";
```
Comment (htran1) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterConfigurationKeys.java]: helix.job.tags -> helixJobTags

File: gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java [hunk 3]
```diff
@@ -257,13 +257,16 @@ protected void executeCancellation() {
...
         ConfigurationKeys.DEFAULT_HELIX_TASK_TIMEOUT_SECONDS) * 1000);
-    
+
     jobConfigBuilder.setFailureThreshold(workUnits.size());
     jobConfigBuilder.addTaskConfigMap(taskConfigMap).setCommand(GobblinTaskRunner.GOBBLIN_TASK_FACTORY_NAME);
     jobConfigBuilder.setNumConcurrentTasksPerInstance(ConfigUtils.getInt(jobConfig,
         GobblinClusterConfigurationKeys.HELIX_CLUSTER_TASK_CONCURRENCY,
         GobblinClusterConfigurationKeys.HELIX_CLUSTER_TASK_CONCURRENCY_DEFAULT));
 
+    List<String> tags = ConfigUtils.getStringList(this.jobConfig, GobblinClusterConfigurationKeys.HELIX_JOB_TAGS_KEY);
```
Comment (htran1) [inline gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java]: There can only be one tag for a job.

**Pull request closed without merging**